### ISSUE TITLE
Research Packit reverse-dependency checks

### ISSUE THREAD

**Issue by lachmanfrantisek** [packit/packit-service#2361]
As part of [the cross-team discussion](https://docs.google.com/document/d/1kr6cZPVTLDr-Y8TF0wD9HDowQH2DlmlCSyRFlgshrxM/edit) about how to help with #2343 , we've come up with an idea how to approach it:

* Use Fedora on GitLab approach for some projects (we can start with Python package(s) since Lumir was part of the discussion and Python would hugely benefit from this functionality).
    * [Ask for a new namespace on GitLab.](https://matrix.to/#/!cfWVeczGVJbiKSlrwi:fedoraproject.org/$E1FGP-pu8czx2HTu1bilWmbd7Ec3X5C2bp3hj31ZnXY?via=fedoraproject.org&via=fedora.im&via=matrix.org)
    * Setup a distro-sync to make some dist-git projects available there .
* Use Packit as a CI on merge-requests targeting these projects. (We can start with supporting existing upstream jobs.)
* Implement the reverse-dependency builds/tests into Packit. (From all possible approaches, this one looked the most promising since we already have the Copr communication in control.)

So, let's try to think about what would it take to do this:
* [ ] Take a look at this approach and write down main tasks and possible issues we might need to resolve if we need to implement this.
* [ ] Write down concerns and riscs of this approach.
* [ ] Think also about possible configuration.
* [ ] Take a look at how we can use Copr for running reverse-dependency builds. (You can get inspired by [mass-prebuild](https://gitlab.com/fedora/packager-tools/mass-prebuild/) or ask Lumir how they are doing this.)
* [ ] Organise a discussion within a team and present the findings.

**Comment by lachmanfrantisek** [comment]
@frenzymadness Good point, Lumír.

A lot is still unclear and that's why I've created this spike card...;)

I don't know many details about the OSCI tooling, but I would say it takes care only of the sync. At least for the start, it would probably be on you, as a maintainer, to disable pull requests on Pagure site if wanted:
![image](https://github.com/packit/packit-service/assets/20214043/b79ee311-0247-4cf8-9301-3dd50c9f4227)

For us, the difference between GitLab/Pagure from the implementation point of view is not the hardest part of the implementation. If the FedoraOnGitLab won't work, we can do the Pagure support. Maybe the GitLab is just more future-proof. The crucial part is to handle these PRs as dist-git content and the reverse-dep logic itself.

**Comment by lachmanfrantisek** [comment]
Maybe just one concern -- how the user rights can be handled on GitLab. If we force people to create a GitLab account and also if the maintainer permissions can be synced.

**Comment by majamassarini** [comment]
@lachmanfrantisek I close this because, if I am not wrong, for the gilab dist-git integration do exist other card, right?

**Comment by lachmanfrantisek** [comment]
@majamassarini I don't think we have a card but definitely independent of your research. (I'll note it in [#2462](https://github.com/packit/packit-service/issues/2462) since it's more about how to do dist-git PRs.
